### DISCUSSION TITLE
How to develop / contribute code changes to consulo?

### DISCUSSION THREAD

**Discussion by TCROC** [consulo/consulo#442] [Q&A] [answered]
Can I open the [consulo repo](https://github.com/consulo/consulo) inside of the [consulo ide](https://consulo.app/) and build and develop it?  I was trying to do this, but whenever I open the consulo project with the consulo ide, the consulo ide ignores it / doesn't do anything.

**Comment by VISTALL** [comment, marked as answer]
You need install few plugins. 

* Consulo DevKit (with dependencies) 
* And Maven if not installed

Install JDK11 +

And then only open any Consulo projects (it will suggest Maven import).

No need build Consulo from scratch.

Just download repo what you need - and open it. 

For start need execute 
 mvn consulo:run-desktop -pl consulo.plugin:PLUGIN_ID (change to plugin what you want)

**Reply by TCROC** [reply]
Is this the "Consulo DevKit"?

![image](https://user-images.githubusercontent.com/24307049/101358974-9bb53e00-3869-11eb-9be3-1e9d9d795106.png)

**Reply by VISTALL** [reply]
yeap

**Reply by TCROC** [reply]
These are my currently installed plugins:

![image](https://user-images.githubusercontent.com/24307049/101359761-bd62f500-386a-11eb-9d19-f8b7ea9130a2.png)

And I'm trying to open the main consulo repo

![image](https://user-images.githubusercontent.com/24307049/101359894-dff50e00-386a-11eb-967b-bac06d4e7adf.png)

But when I do, nothing happens

**Reply by VISTALL** [reply]
Second note.

Maven plugin required.

**Reply by TCROC** [reply]
I got this far now, but the "Next" button isn't working

![image](https://user-images.githubusercontent.com/24307049/101360729-0e271d80-386c-11eb-9a6f-0dc5fc467ec3.png)

**Reply by TCROC** [reply]
Here is the log:

[consulo.log](https://github.com/consulo/consulo/files/5653156/consulo.log)

**Reply by VISTALL** [reply]
Goto to Settings, and select Sdk&Bundles&Tools node, and add JDK. Need fix it

**Reply by TCROC** [reply]
Now I'm this far :).  It appears to be resolving Maven dependencies.  I'll let you know how that goes

![image](https://user-images.githubusercontent.com/24307049/101361812-788c8d80-386d-11eb-92a1-882b5ee75a0b.png)

**Reply by TCROC** [reply]
It appears to have worked! :)

![image](https://user-images.githubusercontent.com/24307049/101362464-226c1a00-386e-11eb-8bc3-79c6510744dc.png)

**Reply by VISTALL** [reply]
In right top menu - select **sandbox-desktop [11] [MODULES]**

But check JDK settings - must be set to your JDK

**Reply by VISTALL** [reply]
It will be long start

**Reply by TCROC** [reply]
Yep its running for me :)

![image](https://user-images.githubusercontent.com/24307049/101378028-91eb0500-3880-11eb-952d-cc80a50c93a8.png)

I ran the mvn install last night so it actually didn't take too long to start up.  Maybe a couple minutes the first time.
